src/fsharp/Optimizer.fs

```diff
@@ -3090,7 +3090,7 @@ and OptimizeBinding cenv isRec env (TBind(vref, expr, spBind)) =
                     | None -> false
                     | Some mbrTyconRef ->
                     // Check we can deref system_MarshalByRefObject_tcref. When compiling against the Silverlight mscorlib we can't
-                    if mbrTyconRef.TryDeref.IsSome then
+                    if ValueOption.isSome mbrTyconRef.TryDeref then
```

Comment:
Please, we should have defined an `.IsSome` property on `ValueOption` before accepting this change.

Part of the whole point of property syntax is that it creates far less churn when changing type.